Implement MongoDB schema for user authentication data

kathinka opened the pull request:
Added model for User and travel - added some suggestions for optional entries and security.

also:
Removed unused and decrepitated crypto dependency from backend package

vittoriamatteoli commented on backend/models/travel-model.js:
```diff
@@ -0,0 +1,36 @@
+    type: Date,
+    required: [true, "Date is required"],
+  },
+  startLocation: {
+    type: String,
+    required: [true, "Start location is required"],
+  },
+  endLocation: {
+    type: String,
+    required: [true, "End location is required"],
+  },
```
Are this 2 values to store the history of travels in the Dashboard ?

kathinka replied:
Its for storing the start and end of a route - so we can use it for multiple things:
 if we want to use this data later for 
 calculation, statistics or maybe show popular routes etc.
 
Now we store it on each travelroute the user preforms - so yes! this will let us show detailed info for previous routes in our dash.

El1an3 commented on backend/models/travel-model.js:
```diff
@@ -0,0 +1,36 @@
+
+
+// added distance and date to the travelSchema as required fields so we can use the data to calculate the carbon footprint/pollution of the travel or points better. or to calculate the total distance traveled by the user.
+//date seemed like a good idea to add to the travelSchema so we can use it to calculate the total distance traveled by the user in a specific time period. or just for sorting the travels by date for instance.
```
yes, useful

El1an3 reviewed (approved):
Looks good to me.

kathinka commented:
Merging schema after approval  
Nip tucks on approved pr:

1. Added role to the user schema so we can use authorization in our project as well.
2. Adjusted the token to be a better fit for storing "remember me" tokens. (so a refresh token) (but it is commented out in case we  all dont want it after all.)
